preact.js

```diff
@@ -0,0 +1,39 @@
+var Preact = require('preact')
+
+var Provider = (function (superclass) {
+  function Provider () {
+    superclass.apply(this, arguments)
+  }
+  /* eslint-disable no-proto */
+  if (superclass) Provider.__proto__ = superclass
+  Provider.prototype = Object.create(superclass && superclass.prototype)
+  Provider.prototype.constructor = Provider
+  Provider.prototype.getChildContext = function getChildContext () {
+    return { atom: this.props.atom }
+  }
+  Provider.prototype.render = function render (props) {
+    return props.children[0]
+  }
+  return Provider
+}(Preact.Component))
+
+function connect (mapState, mapActions) {
+  return function connectComponent (Component) {
+    return function (props, context) {
+      var atom = context.atom
+      props = Object.assign(
+        { atom: atom.get(), split: atom.split },
```

Comment:
not the biggest fan of `atom: atom.get()`. if i use a variable called atom everywhere else, it's the atom instance which has `get`/`split` etc. but now it's suddenly my actual state?

Which `mapState`/`mapActions`, they shouldn't even use this directly tbh, can we leave it out? If they _really_   need it, they can get atom from the context

Reply:
I will rename `atom`->`state` here. Also gonna combine mapState and mapActions into one.